File: backend/src/services/imageService.js
const sharp = require('sharp');
const path = require('path');
const fs = require('fs');

async function resizePhoto(inputPath) {
  const dir  = path.dirname(inputPath);
  const base = path.basename(inputPath, path.extname(inputPath));
  const outputPath = path.join(dir, `${base}.jpg`);

  await sharp(inputPath)
    .resize(300, 300, { fit: 'cover', position: 'center' })
    .jpeg({ quality: 85 })
    .toFile(outputPath);

  if (outputPath !== inputPath && fs.existsSync(inputPath)) {
    fs.unlinkSync(inputPath);
  }

  return outputPath;
}

module.exports = { resizePhoto };

File: backend/src/services/image_service.py
import os
from PIL import Image, ImageOps


def resize_photo(input_path: str) -> str:
    base, _ = os.path.splitext(input_path)
    output_path = base + ".jpg"

    with Image.open(input_path) as img:
        resized = ImageOps.fit(img.convert("RGB"), (300, 300), Image.LANCZOS)
        resized.save(output_path, "JPEG", quality=85)

    if output_path != input_path and os.path.exists(input_path):
        os.unlink(input_path)

    return output_path
